File: apps/android/core/model/src/main/kotlin/com/scribblefit/core/model/ExercisePerformanceInsight.kt
package com.scribblefit.core.model

import kotlinx.serialization.Serializable

@Serializable
data class ExercisePerformanceInsight(
    val estimated1RM: Float,
    val prDetected: Boolean,
    val trendDirection: TrendDirection,
    val breakdownText: String
)

File: apps/ios/ScribbleFit/LocalPackages/Core/CoreModel/Sources/CoreModel/ExercisePerformanceInsight.swift
import Foundation

public struct ExercisePerformanceInsight: Sendable, Equatable {
    public let estimated1RM: Float
    public let prDetected: Bool
    public let trendDirection: TrendDirection
    public let breakdownText: String
    
    public init(estimated1RM: Float, prDetected: Bool, trendDirection: TrendDirection, breakdownText: String) {
        self.estimated1RM = estimated1RM
        self.prDetected = prDetected
        self.trendDirection = trendDirection
        self.breakdownText = breakdownText
    }
}
